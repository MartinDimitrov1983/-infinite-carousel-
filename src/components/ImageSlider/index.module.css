.imageWrapper {
    display: flex;
    height: 80%;
    transition: transform 0.3s ease;
  }
  