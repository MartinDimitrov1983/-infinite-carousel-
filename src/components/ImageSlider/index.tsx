import React from 'react';
import { Image } from '../';
import styles from './index.module.css';

interface ImageSliderProps {
  images: string[];
  currentImageIndex: number;
}

const ImageSlider: React.FC<ImageSliderProps> = ({
  images,
  currentImageIndex,
}) => (
  <div className={styles.imageWrapper}>
    {images.map((image, index) => (
      <Image
        key={index}
        src={image}
        alt={`image-${index}`}
        isActive={index === currentImageIndex}
      />
    ))}
  </div>
);

export default ImageSlider;
