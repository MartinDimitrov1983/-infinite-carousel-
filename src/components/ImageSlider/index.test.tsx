import React from 'react';
import { render } from '@testing-library/react';

import ImageSlider from './index';

test('Snapshot Input', () => {
  const { asFragment } = render(
    <ImageSlider
      images={[
        'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRJgbhoPzhSGlHth9jG-UnNQm3TKFFmGVvrXg&usqp=CAU',
        'https://images.unsplash.com/photo-1503023345310-bd7c1de61c7d?ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxzZWFyY2h8Mnx8aHVtYW58ZW58MHx8MHx8fDA%3D&w=1000&q=80',
      ]}
      currentImageIndex={2}
    />,
  );
  expect(asFragment()).toMatchSnapshot();
});
