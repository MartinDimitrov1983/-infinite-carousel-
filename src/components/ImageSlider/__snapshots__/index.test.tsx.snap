// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Input 1`] = `
<DocumentFragment>
  <div
    class="imageWrapper"
  >
    <img
      alt="image-0"
      class="img"
      src="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRJgbhoPzhSGlHth9jG-UnNQm3TKFFmGVvrXg&usqp=CAU"
      style="display: none;"
    />
    <img
      alt="image-1"
      class="img"
      src="https://images.unsplash.com/photo-1503023345310-bd7c1de61c7d?ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxzZWFyY2h8Mnx8aHVtYW58ZW58MHx8MHx8fDA%3D&w=1000&q=80"
      style="display: none;"
    />
  </div>
</DocumentFragment>
`;
